internal/http/routes/api/v1/bookmarks.go

```diff
@@ -167,3 +194,147 @@ func NewBookmarksPIRoutes(logger *logrus.Logger, deps *config.Dependencies) *Boo
 		deps:   deps,
 	}
 }
+
+// updateCache godoc
+//
+//	@Summary					Update Cache and Ebook on server.
+//	@Tags						Auth
+//	@securityDefinitions.apikey	ApiKeyAuth
+//	@Param						payload	body	updateCachePayload	true "Update Cache Payload"`
+//	@Produce					json
+//	@Success					200	{object}	model.Bookmark
+//	@Failure					403	{object}	nil	"Token not provided/invalid"
+//	@Router						/api/v1/bookmaeks/cache [put]
+func (r *BookmarksAPIRoutes) updateCache(c *gin.Context) {
+	ctx := context.NewContextFromGin(c)
+	if !ctx.UserIsLogged() {
+		response.SendError(c, http.StatusForbidden, nil)
+		return
+	}
+
+	// Get server config
+	logger := logrus.New()
+	cfg := config.ParseServerConfiguration(ctx, logger)
+
+	var payload updateCachePayload
+	if err := c.ShouldBindJSON(&payload); err != nil {
+		response.SendInternalServerError(c)
+		return
+	}
+
+	if err := payload.IsValid(); err != nil {
+		response.SendError(c, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// send request to database and get bookmarks
+	filter := database.GetBookmarksOptions{
+		IDs:         payload.Ids,
+		WithContent: true,
+	}
+
+	bookmarks, err := r.deps.Database.GetBookmarks(c, filter)
+	if len(bookmarks) == 0 {
+		r.logger.WithError(err).Error("Bookmark not found")
+		response.SendError(c, 404, "Bookmark not found")
+		return
+	}
+
+	if err != nil {
+		r.logger.WithError(err).Error("error getting bookmakrs")
+		response.SendInternalServerError(c)
+		return
+	}
+	// TODO: limit request to 20
```

Comment:
**Note:** This can be done in the `updateCachePayload.IsValid()` method.

Reply:
Let's keep it like this for now, since the frontend can only show a limited amount of bookmarks anyway.